test/ProcessTest.php

```diff
@@ -59,6 +65,186 @@ public function testCommandCanRun() {
+            ]);
+            $process->start();
+            $promise = $process->join();
+            $completed = false;
+            $promise->onResolve(function () use (&$completed) { $completed = true; });
+            $this->assertInstanceOf(ResourceInputStream::class, $process->getStdout());
+        });
+    }
+
+    public function testGetStderrIsCustomized() {
+        Loop::run(function () {
+            $process = new Process(self::CMD_PROCESS, null, [], [
+                new ResourceOutputStream(fopen(__DIR__.'/../stream', 'w')),
+            ]);
+            $process->start();
+            $promise = $process->join();
+            $completed = false;
+            $promise->onResolve(function () use (&$completed) { $completed = true; });
+            $this->assertInstanceOf(ResourceInputStream::class, $process->getStderr());
+        });
+    }
+
+    public function testProcessEnvIsValid() {
+        $process = new Process(self::CMD_PROCESS, null, [
+            'env_value'
+        ]);
+        $process->start();
+        $promise = $process->join();
+        $completed = false;
+        $promise->onResolve(function () use (&$completed) { $completed = true; });
+        $this->assertSame('env_value', $process->getEnv()[0]);
+    }
+
+    public function testProcessDeferredIsFailed() {
+        $process = new Process(new \Exception());
+        $this->assertNull($process->start());
+    }
+
+    /**
+     * @expectedException \Error
+     */
+    public function testProcessEnvIsInvalid() {
+        $process = new Process(self::CMD_PROCESS, null, [
+            ['error_value']
+        ]);
+    }
+
+    /**
+     * @expectedException \Amp\Process\StatusError
+     * @expectedExceptionMessage The process has not been started
+     */
+    public function testGetStdinIsStatusError() {
+        $process = new Process(self::CMD_PROCESS, null, []);
+        $process->getStdin();
+    }
+
+    /**
+     * @expectedException \Amp\Process\StatusError
+     * @expectedExceptionMessage The process has not been started
+     */
+    public function testGetStdoutIsStatusError() {
+        $process = new Process(self::CMD_PROCESS, null, []);
+        $process->getStdout();
+    }
+
+    /**
+     * @expectedException \Amp\Process\StatusError
+     * @expectedExceptionMessage The process has not been started
+     */
+    public function testGetStderrIsStatusError() {
+        $process = new Process(self::CMD_PROCESS, null, []);
+        $process->getStderr();
+    }
+
+    /**
+     * @expectedException \Amp\Process\StatusError
+     * @expectedExceptionMessage The process has not been started
+     */
+    public function testProcessCanReset() {
+        $process = new Process(self::CMD_PROCESS);
+        $process->start();
+        $promise = $process->join();
+        $completed = false;
+        $promise->onResolve(function () use (&$completed) { $completed = true; });
```

Comment:
`$completed` is unused here.

Reply:
ping, there's no assertion for `$completed`, please either remove it or add an assertion.